src/Microsoft.AspNet.Mvc.Core/Description/DefaultApiDescriptionProvider.cs

```diff
@@ -450,5 +385,279 @@ private IApiResponseMetadataProvider[] GetResponseMetadataAttributes(ControllerA
+            }
+
+            public ApiParameterContext Context { get; }
+
+            public ParameterDescriptor Parameter { get; }
+
+            // Avoid infinite recursion by tracking properties. 
+            private HashSet<PropertyKey> Visited { get; }
+
+            public void WalkParameter()
+            {
+                var modelMetadata = Context.MetadataProvider.GetMetadataForParameter(
+                    modelAccessor: null,
+                    methodInfo: Context.ActionDescriptor.MethodInfo,
+                    parameterName: Parameter.Name);
+
+                var binderMetadata = Parameter.BinderMetadata;
+                if (modelMetadata != null)
+                {
+                    if (binderMetadata != null)
+                    {
+                        modelMetadata.BinderMetadata = binderMetadata;
+                    }
+
+                    var nameProvider = binderMetadata as IModelNameProvider;
+                    if (nameProvider != null && nameProvider.Name != null)
+                    {
+                        modelMetadata.BinderModelName = nameProvider.Name;
+                    }
+                }
+
+                // Attempt to find a binding source for the parameter
+                //
+                // The default is ModelBinding (aka all default value providers)
+                var source = ApiParameterSource.ModelBinding;
+                if (!Visit(modelMetadata, source, containerName: string.Empty))
+                {
+                    // If we get here, then it means we didn't find a match for any of the model. This means that it's
+                    // likely 'model-bound' in the traditional MVC sense (formdata + query string + route data) and
+                    // doesn't use any IBinderMetadata.
+                    // 
+                    // Add a single 'default' parameter description for the model.
+                    Context.Results.Add(CreateResult(modelMetadata, source, containerName: string.Empty));
+                }
+            }
+
+            private bool Visit(ModelMetadata modelMetadata, ApiParameterSource ambientSource, string containerName)
+            {
+                ApiParameterSource source;
+                if (TryGetSource(modelMetadata, out source))
+                {
+                    // We have a definite answer for this model. This is a greedy source like
+                    // [FromBody] so there's no need to consider properties.
+                    Context.Results.Add(CreateResult(modelMetadata, source, containerName));
+
+                    return true;
+                }
+             
+                if (!modelMetadata.Properties.Any())
+                {
+                    if (source == null)
+                    {
+                        // If there are no properties and no source then we don't know how to bind this.
+                        // Return and let it get potentially merged with the container.
```

Comment:
reworded a little, let me know if you like it better. without a suggestion I can't really improve it more sadly.